arkouda/dataframe.py

```diff
@@ -2706,3 +2708,152 @@ def invert_permutation(perm):
     if (unique(perm).size != perm.size) and (perm.size != rng + 1):
         raise ValueError("The array is not a permutation.")
     return coargsort([perm, arange(perm.size)])
+
+
+@typechecked
+def inner_join_merge(left: DataFrame,
+                     right: DataFrame,
+                     on: str
+                     ) -> DataFrame:
+    """
+    Utilizes the ak.join.inner_join function to return an ak
+    DataFrame object containing only rows that are in both
+    the left and right Dataframes, (based on the "on" param),
+    as well as their associated values.
+
+    Parameters
+    ----------
+    left : DataFrame
+        The Left DataFrame to be joined
+    right : DataFrame
+        The Right DataFrame to be joined
+    on : str
+        The name of the DataFrame column the join is being
+        performed on
+
+    Returns
+    -------
+    ij_ak_df : DataFrame
+        Inner-Joined Arkouda DataFrame
+    """
+
+    ij = inner_join(left[on], right[on])
```

Comment:
completely stylistic, feel free to ignore. but maybe we make this return be 
```python
left_inds, right_inds = inner_join(left[on], right[on])
```
i just feel like it's slightly more readable to have like `left[left_inds]` than `left[ij[0]]`